
# Genome sets for comparative genomics

Tiffany created a tutorial powerpoint to provide advice on setting up a KBase narrative for Ralstonia phylogenomics including insight into the quality control. It's a guide for making additional phylogenomics narratives if you want to. **[Download the tutorial .pptx from this github page](https://github.com/lowepowerlab/protocols/blob/main/resources/KBase_tips_Ralstonia.pptx).** Download button is the ↓ arrow towards the right of the page.

## Narrative: RSSC Phylogenomics
The Lowe-Power lab maintains this narrative with all of the publicly available RSSC genomes.  

Contributors: Tiffany Lowe-Power, Vienna Elmgreen

**Version 2: 600-700 genomes [https://narrative.kbase.us/narrative/189849](https://narrative.kbase.us/narrative/189849)**

Version with only the **genomes of strains in Lowe-Power lab strain collection:** [https://narrative.kbase.us/narrative/199187](https://narrative.kbase.us/narrative/199187)

*Original version (don't use; keep for Legacy) [https://narrative.kbase.us/narrative/123807](https://narrative.kbase.us/narrative/123807)*

**Useful genome sets:**
*	RSSC_for_phylogenomics_IncludesBadQualityGenomes.genomeSet 
    * **Purpose:** place new genomes into a phylogenetic tree with all sequenced RSSC. The “Bad Quality” genomes are ones that are non-ideal for functional genomics analyses. 
*	RSSC_highQuality_v2023-06-24.genomeSet
    * **Purpose:** Tiffany filtered these genomes based on CheckM completeness and contamination. This genomeSet can be used for functional genomics analysis. 

## Narrative: Burkholderiales Species Representatives
Tiffany used GTDB to identify genome representatives of approx 300 species in the Burkholderiales order. All genomes were described as complete on GTDB, but it is possible that there are mistakes. The genomes are named according to GTDB taxonomy when the narrative was created in Summer 2023. 

[https://narrative.kbase.us/narrative/142785](https://narrative.kbase.us/narrative/142785)

Useful genome set: 
* Burkholderiales_speciesReps_HighQuality.genomeSet 

## Narrative: Other Ralstonia phylogenomics
Tiffany maintains this narrative with all of the publicly available Ralstonia genomes that are not RSSC.

[https://narrative.akbase.us/narrative/189428](https://narrative.akbase.us/narrative/189428)

## Narrative: Cupriavidus phylogenomics
Tiffany maintains this narrative with all of the publicly available Cupriavidus genus genomes.

[https://narrative.kbase.us/narrative/142720](https://narrative.kbase.us/narrative/142720)

## Narrative: Pandoraea phylogenomics
Tiffany maintains this narrative with all of the publicly available Pandoraea genus genomes.

[https://narrative.kbase.us/narrative/142975](https://narrative.kbase.us/narrative/142975)

# Transcriptomics

Quorum sensing transcriptome datasets from multiple sources:
* [https://narrative.kbase.us/narrative/153894](https://narrative.kbase.us/narrative/153894)
    * Includes GMI1000 vs. phcA in tomato from Khokhani et al. 
* Perrier phcA transcriptome is in a separative narrative with efpR: https://narrative.kbase.us/narrative/154521 

efpR transcriptome from Perrier et al. 
* [https://narrative.kbase.us/narrative/154521](https://narrative.kbase.us/narrative/154521)

Phyl. IIB-1 strain UY031 in potato stems and rich media from de Pedro Jove 2020 
* [https://narrative.kbase.us/narrative/153897](https://narrative.kbase.us/narrative/153897)

Phyl. I strain RS-SY1 in pepper stems from Du et al 2022. 
* [https://narrative.kbase.us/narrative/153936](https://narrative.kbase.us/narrative/153936)

Phyl. IIB-4 strains IBSBF1503 and UW163 in tomato, banana, and melon stems from Ailloud et al. 2016 
* [https://narrative.kbase.us/narrative/153883](https://narrative.kbase.us/narrative/153883)

Phyl. IV-BDB strain UGMSS-Db01 in culture. (The in planta transcriptome had almost no bacterial reads)
* [https://narrative.kbase.us/narrative/153937](https://narrative.kbase.us/narrative/153937)
